modules/terraform/validate_struct.go

```diff
@@ -0,0 +1,110 @@
+package terraform
+
+import (
+	"fmt"
+	"path"
+	"path/filepath"
+
+	"github.com/gruntwork-io/terratest/modules/collections"
+	"github.com/mattn/go-zglob"
+)
+
+type ValidationOptions struct {
+	// The target directory to recursively search for all Terraform directories (those that contain .tf files)
+	// If you provide RootDir and do not pass entries in either IncludeDirs or ExcludeDirs, then all Terraform directories
+	// From the RootDir, recursively, will be validated
+	RootDir string
+	// If you only want to include certain sub directories, add them as paths relative to RootDir. For example, if the
+	// RootDir is /home/project and you want to only include /home/project/examples, add /home/project/examples here
+	// Note that while the struct requires full paths, you can pass relative paths to the NewValidationOptions function
+	// which will build the full paths based on the supplied RootDir
+	IncludeDirs []string
+	// If you want to explicitly exclude certain sub directories, add them as paths relative to RootDir. For example, if the
+	// RootDir is /home/project and you want to include everything EXCEPT /home/project/modules, add
+	// /home/project/modules to this slice. Note that ExcludeDirs is only considered when IncludeDirs is not passed
+	// Note that while the struct requires full paths, you can pass relative paths to the NewValidationOptions function
+	// which will build the full paths based on the supplied RootDir
+	ExcludeDirs []string
+}
+
+// NewValidationOptions returns a ValidationOptions struct, with override-able sane defaults. Note that the
+// ValidationOptions's fields IncludeDirs and ExcludeDirs must be full paths, but this method will accept relative paths
+// and build the full paths when instantiating the ValidationOptions struct,  making it the preferred means of configuring
+// ValidationOptions.
+//
+// For example, if your RootDir is /home/project/ and you want to exclude "modules" and "test" you need
+// only pass the relative paths in your excludeDirs slice like so:
+// opts, err := NewValidationOptions("/home/project", []string{}, []string{"modules", "test"})
+func NewValidationOptions(rootDir string, includeDirs, excludeDirs []string) (ValidationOptions, error) {
+	vo := ValidationOptions{
+		RootDir:     "",
+		IncludeDirs: []string{},
+		ExcludeDirs: []string{},
+	}
+
+	if rootDir == "" {
+		return vo, ValidationUndefinedRootDirErr{}
+	}
+
+	vo.RootDir = rootDir
+
+	if len(includeDirs) > 0 {
+		vo.IncludeDirs = buildFullPathsFromRelative(vo.RootDir, includeDirs)
+	}
+
+	if len(excludeDirs) > 0 {
+		vo.ExcludeDirs = buildFullPathsFromRelative(vo.RootDir, excludeDirs)
+	}
+
+	return vo, nil
+}
+
+func buildFullPathsFromRelative(rootDir string, relativePaths []string) []string {
+	var fullPaths []string
+	for _, relativePath := range relativePaths {
+		fullPaths = append(fullPaths, filepath.Join(rootDir, relativePath))
```

Comment:
Check if the relative path is, in fact, relative, with `filepath.IsAbs`. If it's already an absolute path, don't change it.

Reply:
Done in a1e0206